### PR TITLE
Update/migrate to circle2.0

### PR DESCRIPTION
**Pull request by brennan**
- Removes circle.yml file
- Adds .circleci/config.yml file to migrate CircleCi to v2.0
- Runs ./gradlew fmt (resulting in two line changes in the main lib)

### CODE REVIEW

File: .buildscript/deploy_snapshot.sh [hunk 1]
```diff
@@ -16,7 +16,6 @@ if [ "$CIRCLE_PROJECT_REPONAME" != "$REPO" ]; then
 elif [ "$CIRCLE_PROJECT_USERNAME" != "$USERNAME" ]; then
   echo "Skipping snapshot deployment: wrong owner. Expected '$USERNAME' but was '$CIRCLE_PROJECT_USERNAME'."
 elif [ "$CIRCLE_JDK_VERSION" != "$JDK" ]; then
-  # $CIRCLE_JDK_VERSION must be manually set in circle.yml
```
Comment (f2prateek) [inline .buildscript/deploy_snapshot.sh]: Oh interesting, I missed this.

Looks like even analytics-android failed. https://circleci.com/gh/segmentio/analytics-android/1008

Comment (f2prateek) [reply .buildscript/deploy_snapshot.sh]: Probably ok for now as we don't need snapshots here, but let's hold off on the other integrations until then.

Comment (f2prateek) [reply .buildscript/deploy_snapshot.sh]: Tracking in https://segment.atlassian.net/browse/LIB-160

File: src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java [hunk 2]
```diff
@@ -88,7 +89,8 @@ public String key() {
       config.trackingServer = settings.getString("heartbeatTrackingServer");
       config.channel = settings.getString("heartbeatChannel");
       // default app version to 0.0 if not otherwise present b/c Adobe requires this value
-      config.appVersion = (!isNullOrEmpty(context.getPackageName())) ? context.getPackageName() : "0.0";
+      config.appVersion =
```
Comment (f2prateek) [inline src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java]: I think when ternary lines get too long, this is more readable as a a simple if else.

Comment (brennan) [reply src/main/java/com/segment/analytics/android/integrations/adobeanalytics/AdobeIntegration.java]: Totally - already made that change in another branch. This branch sprouted off of master before I merged the branch with the if/else statement into master. Thanks!

**Pull request merged**